src/System.Private.CoreLib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -920,6 +926,12 @@ internal static TimeSpan GetDateTimeNowUtcOffsetFromUtc(DateTime time, out bool
                         default(TransitionTime),
                         baseUtcDelta,
                         noDaylightTransitions: true);
+
+                if (!IsValidAdjustmentRuleOffest(timeZoneBaseUtcOffset, r))
+                {
+                    NormalizeAdjustmentRuleOffset(timeZoneBaseUtcOffset, ref r);
```

Comment:
>NormalizeAdjustmentRuleOffset [](start = 20, length = 29)

just skip the rule if it is invalid but don't normalize it..

Reply:
that may cause really weird problems - while removing the historical adjustments is ok then removing adjustments in between of other changes may cause quite a bit of confusion